src/OpenApi/src/Microsoft.AspNetCore.OpenApi.csproj

```diff
@@ -7,6 +7,10 @@
     <!-- Needed to support compiling Utf8BufferTextWriter implementation shared with SignalR -->
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Description>Provides APIs for annotating route handler endpoints in ASP.NET Core with OpenAPI annotations.</Description>
+    <IsTrimmable>true</IsTrimmable>
+    <!-- Required to generated native AoT friendly `MapOpenApi` endpoint. -->
+    <EnableRequestDelegateGenerator>true</EnableRequestDelegateGenerator>
```

Comment:
Is this needed? I would have thought the user just needs to publish their app as AOT and it will automatically turn RDG on?

Does adding this here make it always use RDG for OpenAPI now regardless of AOT publish?

Reply:
Yeah, this will source generate the OpenAPI-endpoint regardless of whether trimming is enabled. I believe that's required if we want to make the package AoT-friendly regardless of what the app referencing it is set for.

Reply:
Something seems off. I thought trimming and AOT worked at app build time, so it shouldn't matter how the dll in the nupkg is built.

I remember trimming got turned into "full" by default in 8.0, which means it trims all dependencies now instead of just your app code. And I assume AOT does a similar thing.

Reply:
Yeah, we could just annotate the `MapOpenApi` method here and let the PublishAoT=true flag in the users project handle the source generated compilation.

I personally didn't see an issue with source generating the endpoint directly in the library. It provides a small benefit of more debuggable code for all users and isn't that much different from us writing a RequestDelegate implementation directly in the library considering its only one endpoint.

@eerhardt Thoughts on this? Is it inappropriate to eagerly enable RDG when we compile the library like this instead of annotating the `MapOpenApi` endpoint with `RUC/RDC`?

Reply:
I have no issues with us dog-fooding our own source generators. I think we got a lot of good feedback when @halter73 tried using RDG in the Identity code.

>  I thought trimming and AOT worked at app build time, so it shouldn't matter how the dll in the nupkg is built.

It matters because we ourselves are calling MapGet:

https://github.com/dotnet/aspnetcore/blob/a0652e0852f1dc9ea199a453efd421f6dfc04f1b/src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs#L28-L31

There's no way for the source generator in a user's build to re-write the above code. It has to happen when the above code is compiled.